wasm/src/debug_service.rs

```diff
@@ -31,17 +106,32 @@ impl DebugService {
         }
     }
 
-    pub fn load_source(
+    pub async fn load_source(
         &mut self,
         path: String,
         source: String,
         target_profile: String,
         entry: Option<String>,
+        read_file: ReadFileCallback,
+        list_directory: ListDirectoryCallback,
+        get_manifest: GetManifestCallback,
     ) -> String {
-        let source_map = SourceMap::new(
-            [(path.into(), source.into())],
-            entry.as_deref().map(|value| value.into()),
-        );
+        let loader = DebugServiceProjectLoader::new(read_file, list_directory, get_manifest);
+        let manifest = loader.get_manifest(&path).await;
```

Comment:
Does it make sense to avoid taking in the source contents and instead have a `try_load_project` that takes an `impl AsRef<Path>`.  Then if the project has a manifest, you try to return the project. Otherwise we create a default manifest and load the supplied path as the input file (if it is a file) returning it as a project. Then all code always assumes a project and all project logic stays in `qsc_project`. Something like this (not real code):
```rust
    fn try_load_project<T: AsRef<Path>>(&self, input: T) -> miette::Result<Project> {
        if let Some(manifest) = Manifest::load_from_path(input.as_ref())? {
            self.load_project(manifest)
        } else {
            let manifest = ManifestDescriptor {
                manifest: Manifest {
                    ..Default::default()
                },
                manifest_dir: input.as_ref().to_path_buf(),
            };
            self.load_project(manifest)
        }
    }
```